cs/src/comm/epoxy-transport/EpoxyTlsConfig.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Bond.Comm.Epoxy
+{
+    using System;
+    using System.Net.Security;
+    using System.Security.Authentication;
+    using System.Security.Cryptography.X509Certificates;
+
+    /// <summary>
+    /// Represents join client- and server-side TLS options.
+    /// </summary>
+    public abstract class EpoxyTlsConfig
+    {
+        /// <summary>
+        /// Creates a new instance of the <see cref="EpoxyTlsConfig"/> class.
+        /// </summary>
+        /// <param name="remoteCertificateValidationCallback">
+        /// Optional delegate responsible for validating remote certificates.
+        /// May be <c>null</c> to use the default validation.
+        /// </param>
+        /// <param name="encryptionPolicy">
+        /// The encryption policy to use. Defaults to
+        /// <see cref="System.Net.Security.EncryptionPolicy.RequireEncryption">EncryptionPolicy.RequireEncryption</see>
+        /// </param>
+        /// <param name="enabledProtocols">
+        /// The enabled protocols. Defaults to
+        /// <see cref="SslProtocols.Tls12"/>.
+        /// </param>
+        /// <param name="checkCertificateRevocation">
+        /// Whether certificate revocation is checked. Defaults to <c>true</c>.
+        /// </param>
+        protected EpoxyTlsConfig(
+            RemoteCertificateValidationCallback remoteCertificateValidationCallback = null,
+            EncryptionPolicy encryptionPolicy = EncryptionPolicy.RequireEncryption,
+            SslProtocols enabledProtocols = SslProtocols.Tls12,
+            bool checkCertificateRevocation = true)
+        {
+            RemoteCertificateValidationCallback = remoteCertificateValidationCallback;
+            EncryptionPolicy = encryptionPolicy;
+            EnabledProtocols = enabledProtocols;
+            CheckCertificateRevocation = checkCertificateRevocation;
+        }
+
+        /// <summary>
+        /// Gets an optional delegate responsible for validating remote
+        /// certificates.
+        /// </summary>
+        public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; }
+
+        /// <summary>
+        /// Gets the encryption policy to use.
+        /// </summary>
+        public EncryptionPolicy EncryptionPolicy { get; }
```

Comment:
Why expose this at all? The values of the enum other than the RequireEncryption just seem like easy ways to accidentally talk in the clear.